src/services/impl/write/UserService.js

```diff
@@ -0,0 +1,158 @@
+import BaseService from './../../base/BaseService';
+import UserSchema from './../../../schemas/UserSchema';
+import { USER_PROFILE_STATUS } from '../../../constants';
+import config from './../../../config';
+
+class UserService extends BaseService {
+
+  constructor(options = { scoped: true }) { 
+    super(UserSchema, options);
+  }
+
+  async createUserProfile({
+    username,
+    tenant,
+    signUpPubKey,
+    status,
+    email,
+    attributes,
+    roles
+  }) {
+
+    const result = await this.createOne({
+      _id: username,
+      signUpPubKey: signUpPubKey,
+      status: status,
+      email: email,
+      attributes: attributes,
+      tenant: tenant,
+      roles: roles
+    });
+
+    return result;
+  }
+
+  async updateUserProfile(username, {
+    status,
+    email,
+    attributes,
+  }) {
+
+    const result = await this.updateOne({ _id: username }, {
+      status,
+      email,
+      attributes
+    });
+
+    return result;
+  }
+
+  async deleteUserProfile(username) {
+    const result = await this.deleteOne({ _id: username })
+    return result;
+  }
+
+  async getUser(username) {
+    const profile = await this.findOne({ _id: username, status: USER_PROFILE_STATUS.APPROVED });
+    return profile;
+  }
+
+  // async createUserAccount({ username, pubKey, role }) {
```

Comment:
Do not comment deprecated methods, just throw an exception at the first line of the method's body